src/nvim/screen.c

```diff
@@ -2709,32 +2709,26 @@ win_line (
               c_extra = ' ';
               c_final = NUL;
               char_attr = win_hl_attr(wp, HLF_SC);
-              n_extra = win_signcol_width(wp);
+              n_extra = sign_cells;
 
               if (row == startrow + filler_lines && filler_todo <= 0) {
                   text_sign = buf_getsigntype(wp->w_buffer, lnum, SIGN_TEXT,
                                               sign_idx, count);
                   if (text_sign != 0) {
                       p_extra = sign_get_text(text_sign);
-                      int symbol_blen = (int)STRLEN(p_extra);
                       if (p_extra != NULL) {
                           c_extra = NUL;
-                          c_final = NUL;
-                          // symbol(s) bytes + (filling spaces) (one byte each)
-                          n_extra = symbol_blen +
-                            (win_signcol_width(wp) - mb_string2cells(p_extra));
-                          memset(extra, ' ', sizeof(extra));
-                          STRNCPY(extra, p_extra, STRLEN(p_extra));
-                          p_extra = extra;
-                          p_extra[n_extra] = NUL;
+                          n_extra = (int)STRLEN(p_extra);
+                          sign_cells -= mb_string2cells(p_extra);
```

Comment:
The previous use of `extra` was not clear to me.  Using just `p_extra` and `n_extra` seems to be good, no?
/cc @teto (via 0848add48).